core/src/main/scala/codesearch/core/util/Helper.scala

```diff
@@ -38,31 +39,49 @@ object Helper {
   def readFileAsync(path: String): IO[List[String]] =
     IO(Source.fromFile(path, "UTF-8")).bracket(source => IO.pure(source.getLines.toList))(source => IO(source.close()))
 
-  def hideSymbols(str: String): String = {
-    str.foldRight("") {
-      case (c, res) if SPECIAL_CHARS contains c =>
-        s"\\$c$res"
-      case (c, res) =>
-        s"$c$res"
+  def preciseMatch(query: String): String = {
+    val queryTokens: Seq[Token] = Tokenizer.parseStringWithSpecialSymbols(query)
+    val preciseMatch: Seq[Token] = queryTokens.map {
+      case SpecialSymbol(value @ "|") => Escaped(value.charAt(0))
+      case SpecialSymbol(value @ "^") => Escaped(value.charAt(0))
+      case SpecialSymbol(value @ "$") => Escaped(value.charAt(0))
+      case SpecialSymbol(value @ "+") => Escaped(value.charAt(0))
+      case SpecialSymbol(value @ "*") => Escaped(value.charAt(0))
+      case SpecialSymbol(value @ "(") => Escaped(value.charAt(0))
+      case SpecialSymbol(value @ ")") => Escaped(value.charAt(0))
+      case SpecialSymbol(value @ ".") => Escaped(value.charAt(0))
+      case SpecialSymbol(value @ "?") => Escaped(value.charAt(0))
+      case other @ _ => other
     }
+
+    StringAssembler.buildStringFromTokens(preciseMatch)
+  }
+
+  def spaceInsenstive(query: String): String = {
```

Comment:
Redundant function. `SpaceInsensitive#spaceInsensitiveString` has the same functionality.

Reply:
Fix name?

Reply:
Use `SpaceInsensitive#spaceInsensitiveString` instead `Helper#spaceInsenstive`.

Reply:
Ok!